File: 1338 reduce-array-size-to-the-half/solution.java
class Solution {
    public int minSetSize(int[] arr) {
       int fr[]=new int[100007];
       for(int i:arr) fr[i]++;
       int[] l1=new int[arr.length];
       int k=0;
       for(int i:fr)
        if(i>0) l1[k++]=i;
        Arrays.sort(l1);
        int res=arr.length;
        for(int i=l1.length-1;i>=0;i--){
            res-=l1[i];
            if(res<=arr.length/2) return l1.length-i;
        }
        return l1.length;
    }
}

File: 1338 reduce-array-size-to-the-half/solution.go
func minSetSize(arr []int) int {
    var brr [100007]int
    for _,v:=range arr {
        brr[v]++;
    }
    i := 0
res := []int{}          // start empty
for _, v := range brr {
    if v > 0 {
        res = append(res, v) // dynamically adds element
        i++
    }
}

    sort.Sort(sort.Reverse(sort.IntSlice(res)))

    n:=len(arr)
    for i,v:=range res {
        n-=v
        if n<=len(arr)/2 {
            return i+1
        }
    }
    return len(res)

}
